### ISSUE TITLE
Contract date is incorrect on page contract list

### ISSUE THREAD

**Issue by xsitesoc** [glpi-project/glpi#16890]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.14

### Bug description

We are facing an issue where the contract expiration date is deviating from the actual date when listing the contracts, expires column. When we access the contract, the date information is correct

### Relevant log output

```shell
2024-04-03 14:45:01] glpiphplog.WARNING:   *** PHP User Warning (512): Missing required intl PHP extension in /var/www/html/xsite/src/Session.php at line 747
  Backtrace :
  src/Session.php:747                                trigger_error()
  inc/includes.php:69                                Session::loadLanguage()
  front/cron.php:44                                  include()

[2024-04-03 14:45:01] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "REQUEST_URI" in /var/www/html/xsite/plugins/phpsaml/setup.php at line 145
  Backtrace :
  src/Plugin.php:320                                 plugin_init_phpsaml()
  src/Plugin.php:272                                 Plugin::load()
  inc/includes.php:98                                Plugin->init()
  front/cron.php:44                                  include()

[2024-04-03 14:45:01] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "REQUEST_URI" in /var/www/html/xsite/plugins/phpsaml/setup.php at line 145
  Backtrace :
  src/Plugin.php:320                                 plugin_init_phpsaml()
  src/Plugin.php:272                                 Plugin::load()
  inc/includes.php:98                                Plugin->init()
  front/cron.php:44                                  include()

[2024-04-03 14:45:01] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "REQUEST_URI" in /var/www/html/xsite/plugins/phpsaml/setup.php at line 161
  Backtrace :
  src/Plugin.php:1696                                plugin_post_init_phpsaml()
  src/Plugin.php:276                                 Plugin::doHook()
  inc/includes.php:98                                Plugin->init()
  front/cron.php:44                                  include()

[2024-04-03 14:45:01] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "REQUEST_URI" in /var/www/html/xsite/plugins/phpsaml/setup.php at line 161
  Backtrace :
  src/Plugin.php:1696                                plugin_post_init_phpsaml()
  src/Plugin.php:276                                 Plugin::doHook()
  inc/includes.php:98                                Plugin->init()
  front/cron.php:44                                  include()

[2024-04-03 11:45:33] glpiphplog.WARNING:   *** PHP Warning (2): count(): Parameter must be an array or an object that implements Countable in /var/www/html/xsite/inc/includes.php at line
102
  Backtrace :
  plugins/phpsaml/front/acs.php:12                   include()
  public/index.php:82                                require()

[2024-04-03 11:45:33] glpiphplog.WARNING:   *** PHP Warning (2): count(): Parameter must be an array or an object that implements Countable in /var/www/html/xsite/inc/includes.php at line
158
  Backtrace :
  plugins/phpsaml/front/acs.php:12                   include()
  public/index.php:82                                require()

[2024-04-03 14:46:02] glpiphplog.WARNING:   *** PHP User Warning (512): Missing required intl PHP extension in /var/www/html/xsite/src/Session.php at line 747
  Backtrace :
  src/Session.php:747                                trigger_error()
  inc/includes.php:69                                Session::loadLanguage()
  front/cron.php:44                                  include()
```

### Page URL

https://example.com/front/contract.php

### Steps To reproduce

We changed the contract expiration date, when leaving the contract page and going to the contract list page, the expiration date is not correct

### Your GLPI setup information

<details><summary>Instruções de instalação e configuração</summary><pre>GLPI 10.0.14 ( => /var/www/html/xsite)
Installation mode: TARBALL
Current language:pt_BR

</pre></details><details><summary>Server</summary><pre> 
Operating system: Linux GLPI 5.15.0-1051-aws #​56~20.04.1-Ubuntu SMP Tue Nov 28 15:43:31 UTC 2023 x86_64
PHP 7.4.3-4ubuntu2.20 apache2handler (Core, FFI, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apc, apcu,
	bz2, calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imagick, imap, intl, json, ldap,
	libxml, mbstring, memcache, mysqli, mysqlnd, openssl, pcre, pdo_mysql, posix, pspell, readline, session, shmop, sockets, sodium,
	standard, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="30" memory_limit="1024M" post_max_size="8M" safe_mode="" session.save_handler="files"
	upload_max_filesize="10M"
	disable_functions="pcntl_alarm,pcntl_fork,pcntl_waitpid,pcntl_wait,pcntl_wifexited,pcntl_wifstopped,pcntl_wifsignaled,pcntl_wifcontinued,pcntl_wexitstatus,pcntl_wtermsig,pcntl_wstopsig,pcntl_signal,pcntl_signal_get_handler,pcntl_signal_dispatch,pcntl_get_last_error,pcntl_strerror,pcntl_sigprocmask,pcntl_sigwaitinfo,pcntl_sigtimedwait,pcntl_exec,pcntl_getpriority,pcntl_setpriority,pcntl_async_signals,pcntl_unshare,"
	
Software: Apache ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36
Server Software: (Ubuntu)
	Server Version: 8.0.36-0ubuntu0.20.04.1
	Server SQL Mode: STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION
	Parameters: glpi_xsite@127.0.0.1/glpi_xsite
	Host info: 127.0.0.1 via TCP/IP
	
PHP version (7.4.3-4ubuntu2.20) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, filter, libxml, json, simplexml, xmlreader, xmlwriter.
curl extension is installed.
gd extension is installed.
intl extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (8.0.36) is supported.
No files from previous GLPI version detected.
The log file has been created successfully.
Write access to /var/www/html/xsite/files/_cache has been validated.
Write access to /var/www/html/xsite/files/_cron has been validated.
Write access to /var/www/html/xsite/files has been validated.
Write access to /var/www/html/xsite/files/_dumps has been validated.
Write access to /var/www/html/xsite/files/_graphs has been validated.
Write access to /var/www/html/xsite/files/_lock has been validated.
Write access to /var/www/html/xsite/files/_pictures has been validated.
Write access to /var/www/html/xsite/files/_plugins has been validated.
Write access to /var/www/html/xsite/files/_rss has been validated.
Write access to /var/www/html/xsite/files/_sessions has been validated.
Write access to /var/www/html/xsite/files/_tmp has been validated.
Write access to /var/www/html/xsite/files/_uploads has been validated.

Web server root directory configuration seems safe.
Sessions configuration is secured.
OS and PHP are relying on 64 bits integers.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
Following extensions are installed: bz2, Phar, zip.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to /var/www/html/xsite/marketplace has been validated.
Timezones seems loaded in database.

</pre></details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: "/var/www/html/xsite"
GLPI_CONFIG_DIR: "/var/www/html/xsite/config"
GLPI_VAR_DIR: "/var/www/html/xsite/files"
GLPI_MARKETPLACE_DIR: "/var/www/html/xsite/marketplace"
GLPI_USE_CSRF_CHECK: "1"
GLPI_CSRF_EXPIRES: "7200"
GLPI_CSRF_MAX_TOKENS: "100"
GLPI_USE_IDOR_CHECK: "1"
GLPI_IDOR_EXPIRES: "7200"
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: false
GLPI_SERVERSIDE_URL_ALLOWLIST: ["/^(https?|feed):\\/\\/[^@:]+(\\/.*)?$/"]
GLPI_TELEMETRY_URI: "https://telemetry.glpi-project.org"
GLPI_INSTALL_MODE: "TARBALL"
GLPI_NETWORK_MAIL: "glpi@teclib.com"
GLPI_NETWORK_SERVICES: "https://services.glpi-network.com"
GLPI_MARKETPLACE_ALLOW_OVERRIDE: true
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: true
GLPI_USER_AGENT_EXTRA_COMMENTS: ""
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: "1"
GLPI_AJAX_DASHBOARD: "1"
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: "0"
GLPI_CENTRAL_WARNINGS: "1"
GLPI_TEXT_MAXSIZE: "4000"
GLPI_DOC_DIR: "/var/www/html/xsite/files"
GLPI_CACHE_DIR: "/var/www/html/xsite/files/_cache"
GLPI_CRON_DIR: "/var/www/html/xsite/files/_cron"
GLPI_DUMP_DIR: "/var/www/html/xsite/files/_dumps"
GLPI_GRAPH_DIR: "/var/www/html/xsite/files/_graphs"
GLPI_LOCAL_I18N_DIR: "/var/www/html/xsite/files/_locales"
GLPI_LOCK_DIR: "/var/www/html/xsite/files/_lock"
GLPI_LOG_DIR: "/var/www/html/xsite/files/_log"
GLPI_PICTURE_DIR: "/var/www/html/xsite/files/_pictures"
GLPI_PLUGIN_DOC_DIR: "/var/www/html/xsite/files/_plugins"
GLPI_RSS_DIR: "/var/www/html/xsite/files/_rss"
GLPI_SESSION_DIR: "/var/www/html/xsite/files/_sessions"
GLPI_TMP_DIR: "/var/www/html/xsite/files/_tmp"
GLPI_UPLOAD_DIR: "/var/www/html/xsite/files/_uploads"
GLPI_INVENTORY_DIR: "/var/www/html/xsite/files/_inventories"
GLPI_NETWORK_REGISTRATION_API_URL: "https://services.glpi-network.com/api/registration/"
GLPI_MARKETPLACE_PLUGINS_API_URI: "https://services.glpi-network.com/api/marketplace/"
GLPI_I18N_DIR: "/var/www/html/xsite/locales"
GLPI_VERSION: "10.0.14"
GLPI_SCHEMA_VERSION: "10.0.14@8f13530f7cf0fb07add30ae20013f0374956795b"
GLPI_MARKETPLACE_PRERELEASES: false
GLPI_MIN_PHP: "7.4.0"
GLPI_MAX_PHP: "8.4.0"
GLPI_YEAR: "2024"

</pre></details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.14 in (/var/www/html/xsite/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.8.0 in (/var/www/html/xsite/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.8 in (/var/www/html/xsite/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.6.2 in (/var/www/html/xsite/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/html/xsite/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/xsite/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/xsite/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/xsite/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/xsite/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/xsite/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/xsite/vendor/sabre/vobject/lib)
laminas/laminas-i18n in (/var/www/html/xsite/vendor/laminas/laminas-i18n/src)
laminas/laminas-servicemanager in (/var/www/html/xsite/vendor/laminas/laminas-servicemanager/src)
monolog/monolog in (/var/www/html/xsite/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/xsite/vendor/sebastian/diff/src)
donatj/phpuseragentparser in (/var/www/html/xsite/vendor/donatj/phpuseragentparser/src/UserAgent)
elvanto/litemoji in (/var/www/html/xsite/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/xsite/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/xsite/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/xsite/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/xsite/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/xsite/vendor/rlanvin/php-rrule/src)
ramsey/uuid in (/var/www/html/xsite/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/xsite/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/xsite/vendor/psr/simple-cache/src)
psr/cache in (/var/www/html/xsite/vendor/psr/cache/src)
league/csv in (/var/www/html/xsite/vendor/league/csv/src)
mexitek/phpcolors in (/var/www/html/xsite/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/xsite/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/html/xsite/vendor/guzzlehttp/psr7/src)
glpi-project/inventory_format in (/var/www/html/xsite/vendor/glpi-project/inventory_format/lib/php)
wapmorgan/unified-archive in (/var/www/html/xsite/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/xsite/vendor/paragonie/sodium_compat/src)
symfony/cache in (/var/www/html/xsite/vendor/symfony/cache)
html2text/html2text in (/var/www/html/xsite/vendor/html2text/html2text/src)
symfony/css-selector in (/var/www/html/xsite/vendor/symfony/css-selector)
symfony/dom-crawler in (/var/www/html/xsite/vendor/symfony/dom-crawler)
twig/twig in (/var/www/html/xsite/vendor/twig/twig/src)
twig/string-extra in (/var/www/html/xsite/vendor/twig/string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 in (/var/www/html/xsite/vendor/symfony/polyfill-php80)
symfony/polyfill-php81 in (/var/www/html/xsite/vendor/symfony/polyfill-php81)
symfony/polyfill-php82 in (/var/www/html/xsite/vendor/symfony/polyfill-php82)
league/oauth2-client in (/var/www/html/xsite/vendor/league/oauth2-client/src/Provider)
league/oauth2-google in (/var/www/html/xsite/vendor/league/oauth2-google/src/Provider)
thenetworg/oauth2-azure in (/var/www/html/xsite/vendor/thenetworg/oauth2-azure/src/Provider)
phpCas version 1.3.8 in (/usr/share/php/CAS/source)

</pre></details><details><summary>SQL replicas</summary><pre> 
Not active

</pre></details><details><summary>Notifications</summary><pre> 
Way of sending emails: SMTP+TLS (anonymous@xsite-com-br.mail.protection.outlook.com)

</pre></details><details><summary>Plugins list</summary><pre> 
	advancedplanning     Name: advancedplanning               Version: 1.1.0      State: Enabled                                 
		Install Method: Marketplace
	brightaiassistant    Name: Bright AI Assistant            Version: 0.0.1      State: Enabled                                 
		Install Method: Marketplace
	fields               Name: Campos adicionais              Version: 1.21.6     State: Enabled                                 
		Install Method: Marketplace
	shellcommands        Name: Comandos de Shell              Version: 4.0.1      State: Enabled                                 
		Install Method: Marketplace
	gantt                Name: gantt                          Version: 1.0.4      State: Enabled                                 
		Install Method: Marketplace
	mod                  Name: GLPI Modifications             Version: 1.1.5      State: Error / to clean                        
		Install Method: Manual
	phpsaml              Name: PHP SAML                       Version: 1.2.1      State: Enabled                                 
		Install Method: Manual
	webresources         Name: Web Resources                  Version: 2.0.3      State: Enabled                                 
		Install Method: Marketplace

</pre></details>

**Comment by trasher** [comment]
Your logs states you missed intl php extension, and some issues comig from a plugin. Try without any plugin, and with all required extension installed.

**Comment by xsitesoc** [comment]
We have been tested without any plugin enabled, but no effect

**Comment by cedric-anne** [comment]
@xsitesoc Since #16719, the displayed expiration date is computed based on renewal frequecy, could it be related ?

**Comment by xsitesoc** [comment]
yes, check more prints

![contract_pagelist](https://github.com/glpi-project/glpi/assets/138019993/f9413798-fb9f-4815-b4d5-572729fa03eb)
![inside_contract](https://github.com/glpi-project/glpi/assets/138019993/c5c67c30-ba41-4d54-83b9-b1cf72a6b0e7)

**Comment by cedric-anne** [comment]
@xsitesoc  #16935 should fix this issue. Could you confirm ?

**Comment by cedric-anne** [comment]
Fixed by #16935
